core/lib/utils/src/serde_wrappers.rs

```diff
@@ -25,28 +25,70 @@ impl<P: Prefix> BytesToHexSerde<P> {
     where
         S: Serializer,
     {
-        // First, serialize to hexadecimal string.
-        let hex_value = format!("{}{}", P::prefix(), hex::encode(value));
+        if serializer.is_human_readable() {
+            // First, serialize to hexadecimal string.
+            let hex_value = format!("{}{}", P::prefix(), hex::encode(value));
 
-        // Then, serialize it using `Serialize` trait implementation for `String`.
-        String::serialize(&hex_value, serializer)
+            // Then, serialize it using `Serialize` trait implementation for `String`.
+            String::serialize(&hex_value, serializer)
+        } else {
+            <[u8]>::serialize(value, serializer)
+        }
     }
 
     pub fn deserialize<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
     where
         D: Deserializer<'de>,
     {
-        let deserialized_string = String::deserialize(deserializer)?;
+        if deserializer.is_human_readable() {
+            let deserialized_string = String::deserialize(deserializer)?;
```

Comment:
This place could theoretically trigger the same problem with `is_human_readable()` set for a surrogate deserializer used for `#[serde(flatten)]` groups, but if it works as is, it can be dealt with later.

Reply:
I don't want to pull in another deserializer just for testing.. if the need for one arises, we can revisit. bincode fails with flatten: https://github.com/bincode-org/bincode/issues/245